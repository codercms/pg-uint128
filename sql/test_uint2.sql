-- Testing uint2

-- Ops block

SELECT 1::uint2 = 1::uint2;
SELECT 1::uint2 = 0::uint2;
SELECT 1::uint2 = 1::int2;
SELECT 1::uint2 = 0::int2;
SELECT 1::int2 = 1::uint2;
SELECT 1::int2 = 0::uint2;
SELECT 1::uint2 = 1::int4;
SELECT 1::uint2 = 0::int4;
SELECT 1::int4 = 1::uint2;
SELECT 1::int4 = 0::uint2;
SELECT 1::uint2 = 1::int8;
SELECT 1::uint2 = 0::int8;
SELECT 1::int8 = 1::uint2;
SELECT 1::int8 = 0::uint2;

SELECT 1::uint2 <> 0::uint2;
SELECT 0::uint2 <> 0::uint2;
SELECT 1::uint2 <> 0::int2;
SELECT 0::uint2 <> 0::int2;
SELECT 1::int2 <> 0::uint2;
SELECT 0::int2 <> 0::uint2;
SELECT 1::uint2 <> 0::int4;
SELECT 0::uint2 <> 0::int4;
SELECT 1::int4 <> 0::uint2;
SELECT 0::int4 <> 0::uint2;
SELECT 1::uint2 <> 0::int8;
SELECT 0::uint2 <> 0::int8;
SELECT 1::int8 <> 0::uint2;
SELECT 0::int8 <> 0::uint2;

SELECT 1::uint2 > 0::uint2;
SELECT 0::uint2 > 1::uint2;
SELECT 1::uint2 > 0::int2;
SELECT 0::uint2 > 1::int2;
SELECT 1::int2 > 0::uint2;
SELECT 0::int2 > 1::uint2;
SELECT 1::uint2 > 0::int4;
SELECT 0::uint2 > 1::int4;
SELECT 1::int4 > 0::uint2;
SELECT 0::int4 > 1::uint2;
SELECT 1::uint2 > 0::int8;
SELECT 0::uint2 > 1::int8;
SELECT 1::int8 > 0::uint2;
SELECT 0::int8 > 1::uint2;

SELECT 0::uint2 < 1::uint2;
SELECT 1::uint2 < 0::uint2;
SELECT 0::uint2 < 1::int2;
SELECT 1::uint2 < 0::int2;
SELECT 0::int2 < 1::uint2;
SELECT 1::int2 < 0::uint2;
SELECT 0::uint2 < 1::int4;
SELECT 1::uint2 < 0::int4;
SELECT 0::int4 < 1::uint2;
SELECT 1::int4 < 0::uint2;
SELECT 0::uint2 < 1::int8;
SELECT 1::uint2 < 0::int8;
SELECT 0::int8 < 1::uint2;
SELECT 1::int8 < 0::uint2;

SELECT 1::uint2 >= 0::uint2;
SELECT 0::uint2 >= 1::uint2;
SELECT 1::uint2 >= 1::uint2;
SELECT 1::uint2 >= 0::int2;
SELECT 0::uint2 >= 1::int2;
SELECT 1::uint2 >= 1::int2;
SELECT 1::int2 >= 0::uint2;
SELECT 0::int2 >= 1::uint2;
SELECT 1::int2 >= 1::uint2;
SELECT 1::uint2 >= 0::int4;
SELECT 0::uint2 >= 1::int4;
SELECT 1::uint2 >= 1::int4;
SELECT 1::int4 >= 0::uint2;
SELECT 0::int4 >= 1::uint2;
SELECT 1::int4 >= 1::uint2;
SELECT 1::uint2 >= 0::int8;
SELECT 0::uint2 >= 1::int8;
SELECT 1::uint2 >= 1::int8;
SELECT 1::int8 >= 0::uint2;
SELECT 0::int8 >= 1::uint2;
SELECT 1::int8 >= 1::uint2;

SELECT 0::uint2 <= 1::uint2;
SELECT 1::uint2 <= 0::uint2;
SELECT 1::uint2 <= 1::uint2;
SELECT 0::uint2 <= 1::int2;
SELECT 1::uint2 <= 0::int2;
SELECT 1::uint2 <= 1::int2;
SELECT 0::int2 <= 1::uint2;
SELECT 1::int2 <= 0::uint2;
SELECT 1::int2 <= 1::uint2;
SELECT 0::uint2 <= 1::int4;
SELECT 1::uint2 <= 0::int4;
SELECT 1::uint2 <= 1::int4;
SELECT 0::int4 <= 1::uint2;
SELECT 1::int4 <= 0::uint2;
SELECT 1::int4 <= 1::uint2;
SELECT 0::uint2 <= 1::int8;
SELECT 1::uint2 <= 0::int8;
SELECT 1::uint2 <= 1::int8;
SELECT 0::int8 <= 1::uint2;
SELECT 1::int8 <= 0::uint2;
SELECT 1::int8 <= 1::uint2;

SELECT 120::uint2 + 10::uint2;
SELECT 120::uint2 + 10::int2;
SELECT 65535::uint2 + 1::int2;
SELECT 120::int2 + 10::uint2;
SELECT 32767::int2 + 1::uint2;
SELECT (-120)::int2 + 10::uint2;
SELECT 120::uint2 + 10::int4;
SELECT 65535::uint2 + 1::int4;
SELECT 120::int4 + 10::uint2;
SELECT 2147483647::int4 + 1::uint2;
SELECT (-120)::int4 + 10::uint2;
SELECT 120::uint2 + 10::int8;
SELECT 65535::uint2 + 1::int8;
SELECT 120::int8 + 10::uint2;
SELECT 9223372036854775807::int8 + 1::uint2;
SELECT (-120)::int8 + 10::uint2;

SELECT 120::uint2 - 10::uint2;
SELECT 120::uint2 - 10::int2;
SELECT 0::uint2 - 1::int2;
SELECT 120::int2 - 10::uint2;
SELECT (-32768)::int2 - 1::uint2;
SELECT (-120)::int2 - 10::uint2;
SELECT 120::uint2 - 10::int4;
SELECT 0::uint2 - 1::int4;
SELECT 120::int4 - 10::uint2;
SELECT (-2147483648)::int4 - 1::uint2;
SELECT (-120)::int4 - 10::uint2;
SELECT 120::uint2 - 10::int8;
SELECT 0::uint2 - 1::int8;
SELECT 120::int8 - 10::uint2;
SELECT (-9223372036854775808)::int8 - 1::uint2;
SELECT (-120)::int8 - 10::uint2;

SELECT 120::uint2 * 10::uint2;
SELECT 120::uint2 * 10::int2;
SELECT 65535::uint2 * 2::int2;
SELECT 120::int2 * 10::uint2;
SELECT 32767::int2 * 2::uint2;
SELECT (-120)::int2 * 10::uint2;
SELECT 120::uint2 * 10::int4;
SELECT 65535::uint2 * 2::int4;
SELECT 120::int4 * 10::uint2;
SELECT 2147483647::int4 * 2::uint2;
SELECT (-120)::int4 * 10::uint2;
SELECT 120::uint2 * 10::int8;
SELECT 65535::uint2 * 2::int8;
SELECT 120::int8 * 10::uint2;
SELECT 9223372036854775807::int8 * 2::uint2;
SELECT (-120)::int8 * 10::uint2;

SELECT 120::uint2 / 10::uint2;
SELECT 120::uint2 / 10::int2;
SELECT 1::uint2 / 0::int2;
SELECT 120::int2 / 10::uint2;
SELECT 1::int2 / 0::uint2;
SELECT (-120)::int2 / 10::uint2;
SELECT 120::uint2 / 10::int4;
SELECT 1::uint2 / 0::int4;
SELECT 120::int4 / 10::uint2;
SELECT 1::int4 / 0::uint2;
SELECT (-120)::int4 / 10::uint2;
SELECT 120::uint2 / 10::int8;
SELECT 1::uint2 / 0::int8;
SELECT 120::int8 / 10::uint2;
SELECT 1::int8 / 0::uint2;
SELECT (-120)::int8 / 10::uint2;

SELECT 120::uint2 % 10::uint2;
SELECT 120::uint2 % 10::int2;
SELECT 1::uint2 % 0::int2;
SELECT 120::int2 % 10::uint2;
SELECT 1::int2 % 0::uint2;
SELECT (-120)::int2 % 10::uint2;
SELECT 120::uint2 % 10::int4;
SELECT 1::uint2 % 0::int4;
SELECT 120::int4 % 10::uint2;
SELECT 1::int4 % 0::uint2;
SELECT (-120)::int4 % 10::uint2;
SELECT 120::uint2 % 10::int8;
SELECT 1::uint2 % 0::int8;
SELECT 120::int8 % 10::uint2;
SELECT 1::int8 % 0::uint2;
SELECT (-120)::int8 % 10::uint2;

-- Agg ops block

SELECT sum(s::uint2) FROM generate_series(1, 10000) s;
SELECT avg(s::uint2) FROM generate_series(1, 10000) s;
SELECT min(s::uint2) FROM generate_series(1, 10000) s;
SELECT max(s::uint2) FROM generate_series(1, 10000) s;

-- Generate series block

SELECT s FROM generate_series(1::uint2, 10::uint2) s;
SELECT s FROM generate_series(1::uint2, 10::uint2, 2::uint2) s;
