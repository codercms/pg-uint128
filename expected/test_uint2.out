-- Testing uint2
-- Ops block
SELECT 1::uint2 = 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 = 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 = 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 = 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 = 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 = 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 = 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 = 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 = 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 = 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 = 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 = 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 = 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 = 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <> 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <> 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <> 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <> 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 <> 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 <> 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <> 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <> 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 <> 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 <> 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <> 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <> 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 <> 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 <> 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 > 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 > 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 > 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 > 1::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 > 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 > 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 > 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 > 1::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 > 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 > 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 > 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 > 1::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 > 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 > 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint2 < 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 < 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint2 < 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 < 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 0::int2 < 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 < 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint2 < 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 < 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 0::int4 < 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 < 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 0::uint2 < 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 < 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 0::int8 < 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 < 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 >= 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 >= 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 >= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 >= 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 >= 1::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 >= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 >= 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 >= 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 >= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 >= 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 >= 1::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 >= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 >= 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 >= 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 >= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 >= 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 >= 1::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 >= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 >= 0::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 >= 1::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 >= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 <= 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 <= 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 <= 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 <= 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 <= 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 0::uint2 <= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::uint2 <= 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::uint2 <= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 <= 0::uint2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 <= 1::uint2;
 ?column? 
----------
 t
(1 row)

SELECT 120::uint2 + 10::uint2;
 ?column? 
----------
 130
(1 row)

SELECT 120::uint2 + 10::int2;
 ?column? 
----------
 130
(1 row)

SELECT 65535::uint2 + 1::int2;
ERROR:  uint2 out of range
SELECT 120::int2 + 10::uint2;
 ?column? 
----------
      130
(1 row)

SELECT 32767::int2 + 1::uint2;
ERROR:  int2 out of range
SELECT (-120)::int2 + 10::uint2;
 ?column? 
----------
     -110
(1 row)

SELECT 120::uint2 + 10::int4;
 ?column? 
----------
 130
(1 row)

SELECT 65535::uint2 + 1::int4;
ERROR:  uint2 out of range
SELECT 120::int4 + 10::uint2;
 ?column? 
----------
      130
(1 row)

SELECT 2147483647::int4 + 1::uint2;
ERROR:  int4 out of range
SELECT (-120)::int4 + 10::uint2;
 ?column? 
----------
     -110
(1 row)

SELECT 120::uint2 + 10::int8;
 ?column? 
----------
 130
(1 row)

SELECT 65535::uint2 + 1::int8;
ERROR:  uint2 out of range
SELECT 120::int8 + 10::uint2;
 ?column? 
----------
      130
(1 row)

SELECT 9223372036854775807::int8 + 1::uint2;
ERROR:  int8 out of range
SELECT (-120)::int8 + 10::uint2;
 ?column? 
----------
     -110
(1 row)

SELECT 120::uint2 - 10::uint2;
 ?column? 
----------
 110
(1 row)

SELECT 120::uint2 - 10::int2;
 ?column? 
----------
 110
(1 row)

SELECT 0::uint2 - 1::int2;
ERROR:  uint2 out of range
SELECT 120::int2 - 10::uint2;
 ?column? 
----------
      110
(1 row)

SELECT (-32768)::int2 - 1::uint2;
ERROR:  int2 out of range
SELECT (-120)::int2 - 10::uint2;
 ?column? 
----------
     -130
(1 row)

SELECT 120::uint2 - 10::int4;
 ?column? 
----------
 110
(1 row)

SELECT 0::uint2 - 1::int4;
ERROR:  uint2 out of range
SELECT 120::int4 - 10::uint2;
 ?column? 
----------
      110
(1 row)

SELECT (-2147483648)::int4 - 1::uint2;
ERROR:  int4 out of range
SELECT (-120)::int4 - 10::uint2;
 ?column? 
----------
     -130
(1 row)

SELECT 120::uint2 - 10::int8;
 ?column? 
----------
 110
(1 row)

SELECT 0::uint2 - 1::int8;
ERROR:  uint2 out of range
SELECT 120::int8 - 10::uint2;
 ?column? 
----------
      110
(1 row)

SELECT (-9223372036854775808)::int8 - 1::uint2;
ERROR:  int8 out of range
SELECT (-120)::int8 - 10::uint2;
 ?column? 
----------
     -130
(1 row)

SELECT 120::uint2 * 10::uint2;
 ?column? 
----------
 1200
(1 row)

SELECT 120::uint2 * 10::int2;
 ?column? 
----------
 1200
(1 row)

SELECT 65535::uint2 * 2::int2;
ERROR:  uint2 out of range
SELECT 120::int2 * 10::uint2;
 ?column? 
----------
     1200
(1 row)

SELECT 32767::int2 * 2::uint2;
ERROR:  int2 out of range
SELECT (-120)::int2 * 10::uint2;
 ?column? 
----------
    -1200
(1 row)

SELECT 120::uint2 * 10::int4;
 ?column? 
----------
 1200
(1 row)

SELECT 65535::uint2 * 2::int4;
ERROR:  uint2 out of range
SELECT 120::int4 * 10::uint2;
 ?column? 
----------
     1200
(1 row)

SELECT 2147483647::int4 * 2::uint2;
ERROR:  int4 out of range
SELECT (-120)::int4 * 10::uint2;
 ?column? 
----------
    -1200
(1 row)

SELECT 120::uint2 * 10::int8;
 ?column? 
----------
 1200
(1 row)

SELECT 65535::uint2 * 2::int8;
ERROR:  uint2 out of range
SELECT 120::int8 * 10::uint2;
 ?column? 
----------
     1200
(1 row)

SELECT 9223372036854775807::int8 * 2::uint2;
ERROR:  int8 out of range
SELECT (-120)::int8 * 10::uint2;
 ?column? 
----------
    -1200
(1 row)

SELECT 120::uint2 / 10::uint2;
 ?column? 
----------
 12
(1 row)

SELECT 120::uint2 / 10::int2;
 ?column? 
----------
 12
(1 row)

SELECT 1::uint2 / 0::int2;
ERROR:  division by zero
SELECT 120::int2 / 10::uint2;
 ?column? 
----------
       12
(1 row)

SELECT 1::int2 / 0::uint2;
ERROR:  division by zero
SELECT (-120)::int2 / 10::uint2;
 ?column? 
----------
        0
(1 row)

SELECT 120::uint2 / 10::int4;
 ?column? 
----------
 12
(1 row)

SELECT 1::uint2 / 0::int4;
ERROR:  division by zero
SELECT 120::int4 / 10::uint2;
 ?column? 
----------
       12
(1 row)

SELECT 1::int4 / 0::uint2;
ERROR:  division by zero
SELECT (-120)::int4 / 10::uint2;
 ?column? 
----------
        0
(1 row)

SELECT 120::uint2 / 10::int8;
 ?column? 
----------
 12
(1 row)

SELECT 1::uint2 / 0::int8;
ERROR:  division by zero
SELECT 120::int8 / 10::uint2;
 ?column? 
----------
       12
(1 row)

SELECT 1::int8 / 0::uint2;
ERROR:  division by zero
SELECT (-120)::int8 / 10::uint2;
 ?column? 
----------
        0
(1 row)

SELECT 120::uint2 % 10::uint2;
 ?column? 
----------
 0
(1 row)

SELECT 120::uint2 % 10::int2;
 ?column? 
----------
 0
(1 row)

SELECT 1::uint2 % 0::int2;
ERROR:  division by zero
SELECT 120::int2 % 10::uint2;
 ?column? 
----------
        0
(1 row)

SELECT 1::int2 % 0::uint2;
ERROR:  division by zero
SELECT (-120)::int2 % 10::uint2;
 ?column? 
----------
     -120
(1 row)

SELECT 120::uint2 % 10::int4;
 ?column? 
----------
 0
(1 row)

SELECT 1::uint2 % 0::int4;
ERROR:  division by zero
SELECT 120::int4 % 10::uint2;
 ?column? 
----------
        0
(1 row)

SELECT 1::int4 % 0::uint2;
ERROR:  division by zero
SELECT (-120)::int4 % 10::uint2;
 ?column? 
----------
     -120
(1 row)

SELECT 120::uint2 % 10::int8;
 ?column? 
----------
 0
(1 row)

SELECT 1::uint2 % 0::int8;
ERROR:  division by zero
SELECT 120::int8 % 10::uint2;
 ?column? 
----------
        0
(1 row)

SELECT 1::int8 % 0::uint2;
ERROR:  division by zero
SELECT (-120)::int8 % 10::uint2;
 ?column? 
----------
     -120
(1 row)

-- Agg ops block
SELECT sum(s::uint2) FROM generate_series(1, 10000) s;
   sum    
----------
 50005000
(1 row)

SELECT avg(s::uint2) FROM generate_series(1, 10000) s;
          avg          
-----------------------
 5000.5000000000000000
(1 row)

SELECT min(s::uint2) FROM generate_series(1, 10000) s;
 min 
-----
 1
(1 row)

SELECT max(s::uint2) FROM generate_series(1, 10000) s;
  max  
-------
 10000
(1 row)

-- Generate series block
SELECT s FROM generate_series(1::uint2, 10::uint2) s;
 s  
----
 1
 2
 3
 4
 5
 6
 7
 8
 9
 10
(10 rows)

SELECT s FROM generate_series(1::uint2, 10::uint2, 2::uint2) s;
 s 
---
 1
 3
 5
 7
 9
(5 rows)

